sqlmesh/core/engine_adapter/base.py

```diff
@@ -481,15 +489,82 @@ def _create_table_from_columns(
             if primary_key and self.SUPPORTS_INDEXES
             else []
         )
-        schema = exp.Schema(
-            this=table,
-            expressions=[
-                exp.ColumnDef(this=exp.to_identifier(column), kind=kind)
-                for column, kind in columns_to_types.items()
-            ]
-            + primary_key_expression,
+
+        column_descriptions = kwargs.pop("column_descriptions", None)
+        schema = self._build_schema_exp(
+            table,
+            columns_to_types,
+            column_descriptions if self.COMMENT_CREATION == CommentCreation.IN_SCHEMA_DEF else None,
+            primary_key_expression,
+        )
+
+        table_description = kwargs.pop("table_description", None)
+        self._create_table(
+            schema,
+            None,
+            exists=exists,
+            columns_to_types=columns_to_types,
+            table_description=table_description
+            if self.COMMENT_CREATION == CommentCreation.IN_SCHEMA_DEF
+            else None,
+            **kwargs,
         )
-        self._create_table(schema, None, exists=exists, columns_to_types=columns_to_types, **kwargs)
+
+        # Register comments with commands if the engine doesn't support comments in the schema or CREATE
+        if (
+            table_description or column_descriptions
+        ) and self.COMMENT_CREATION == CommentCreation.COMMENT_COMMAND_ONLY:
+            self._create_comments(
+                table_name,
+                table_description,
+                column_descriptions,
+                columns_to_types=columns_to_types,
+            )
+
+    def _build_schema_exp(
+        self,
+        table: exp.Table,
+        columns_to_types: t.Dict[str, exp.DataType],
+        column_descriptions: t.Optional[t.Dict[str, str]],
+        expressions: t.List[exp.PrimaryKey] = [],
+    ) -> exp.Schema:
+        """
+        Build a schema expression for a table, columns, column comments, and additional schema properties.
+        """
+        if column_descriptions:
+            return exp.Schema(
+                this=table,
+                expressions=[
+                    exp.ColumnDef(
+                        this=exp.to_identifier(column),
+                        kind=kind,
+                        constraints=self._build_col_comment_exp(column, column_descriptions),
```

Comment:
Is this the only thing that differs between branches? Can we just have a single branch instead and just do:
```
constraints=self._build_col_comment_exp(column, column_descriptions) if column_descriptions else None
```
?